### PR TITLE
#145028641 Front tests 

### PR DESCRIPTION
**Pull request by georgreen**
#### What does this PR do?
- Adds unit tests for the app hc/front
#### Description of Task to be completed?
- Add tests to test team access functionality in channels
- Add tests to check bad kinds of channels do not work
- Add tests to check team access works on checks
- Add tests to check pushover validate priority rejecting invalid ones
- Refactor code base 

#### How should this be manually tested?
- python manage.py test hc/front

#### What are the relevant pivotal tracker stories?
- #145028641

### CODE REVIEW

File: hc/front/views.py [hunk 1]
```diff
@@ -330,6 +331,7 @@ def channel_checks(request, code):
         "channel": channel
     }
 
+    print("returned page")
```
Comment (collinmutembei) [inline hc/front/views.py]: I believe this print statement was introduced for debugging purposes. In which case you should avoid pushing code with debug statements.

**Review by collinmutembei — CHANGES_REQUESTED** [review]
👀 Still looking into this.

**Comment by collinmutembei** [comment]
Remove the square brackets on the PR title.

File: hc/front/views.py (hunk 1 above)
Comment (georgreen) [reply hc/front/views.py]: done

File: hc/front/tests/test_add_channel.py [hunk 2]
```diff
@@ -37,5 +36,30 @@ def test_instructions_work(self):
...
+    def test_team_access_works(self):
+        kinds = ("email", "webhook", "pd", "pushover", "hipchat", "victorops")
+        for kind_name in kinds:
+            self.channel = Channel(user=self.alice, kind=kind_name)
+            self.channel.value = "test that team access works"
+            self.channel.save()
+
+            url = "/integrations/{}/checks/".format(self.channel.code)
+
+            self.client.login(username="bob@example.org", password="password")
```
Comment (collinmutembei) [inline hc/front/tests/test_add_channel.py]: Do lines [51-53](https://github.com/andela/healthchecks-thunknalla/blob/a066e49909c4a8b039d53bebf7047853310afff4/hc/front/tests/test_add_channel.py#L51-L53) need to be in the for loop?

File: hc/front/tests/test_add_channel.py [hunk 3]
```diff
@@ -37,5 +36,30 @@ def test_instructions_work(self):
...
+
+
+# Test that bad kinds don't work//channels that raise Not implemented error
+class BadKindTestCase(BaseTestCase):
+    def test_bad_kind(self):
+        self.client.login(username="alice@example.org", password="password")
+        bad_kinds = ('0798885255', 'twitter', 'instagram', 'whatsapp',
+                     'telegram', 'facebook', 'reddit')
+        for any_kind in bad_kinds:
+            url = "/integrations/add_%s/" % any_kind
```
Comment (collinmutembei) [inline hc/front/tests/test_add_channel.py]: I recommend using `str.format()` for string formatting.

Resource: [pyformat.info](https://pyformat.info)

File: hc/front/tests/test_add_channel.py [hunk 4]
```diff
@@ -37,5 +36,30 @@ def test_instructions_work(self):
...
+            r = self.client.get(url)
+            assert r.status_code == 404
```
Comment (collinmutembei) [inline hc/front/tests/test_add_channel.py]: Replace this assert statement with its equivalent in Django.

Resource: [Django assertions](https://docs.djangoproject.com/en/1.10/topics/testing/tools/#assertions)

File: hc/front/tests/test_add_check.py [hunk 5]
```diff
@@ -12,3 +12,18 @@ def test_it_works(self):
...
+        self.client.login(username="bob@example.org", password="password")
+
+        url = "/checks/%s/timeout/" % self.check.code
+        payload = {"timeout": 5000, "grace": 60}
+
+        r = self.client.post(url, data=payload)
+        self.assertRedirects(r, "/checks/")
+
+        check = Check.objects.get(code=self.check.code)
+        assert check.timeout.total_seconds() == 5000
```
Comment (collinmutembei) [inline hc/front/tests/test_add_check.py]: Use djangoish assert statements. See previous similar comment.

File: hc/front/tests/test_add_check.py [hunk 6]
```diff
@@ -12,3 +12,18 @@ def test_it_works(self):
...
+    def test_team_access(self):
+        self.check = Check(user=self.alice)
+        self.check.save()
+
+        self.client.login(username="bob@example.org", password="password")
+
+        url = "/checks/%s/timeout/" % self.check.code
+        payload = {"timeout": 5000, "grace": 60}
+
+        r = self.client.post(url, data=payload)
```
Comment (collinmutembei) [inline hc/front/tests/test_add_check.py]: Refrain from using variable names such as `r`. Use a descriptive name. If you find such in existing code, try to understand the context and update them to a meaningful name.

@georgreen @davidmukiibi @victoriaaoka @Andretalik @Kachulio1

File: hc/front/tests/test_add_pushbover.py [hunk 7]
```diff
@@ -38,3 +39,15 @@ def test_it_validates_nonce(self):
         assert r.status_code == 403
 
     ### Test that pushover validates priority
+    def test_validate_priority(self):
```
Comment (collinmutembei) [inline hc/front/tests/test_add_pushbover.py]: @georgreen @davidmukiibi Could either of you explain what is going on in this test method.

**Review by collinmutembei — CHANGES_REQUESTED** [review]
@georgreen @davidmukiibi Have a look at the comments I have left and attend to them.

File: hc/front/tests/test_add_pushbover.py (hunk 7 above)
Comment (georgreen) [reply hc/front/tests/test_add_pushbover.py]: @collinmutembei pushover is a service that provides a simple way to send notifications to devices from our web app, the notification sent have priority ranging from -2 to 2, this function test that we can only add pushover notification in this valid range so anything else should not be validated ie strings and integers out of range. If the user tries to add such a pushover without the correct priority the response status code should be 400 (bad request). Hope am clear enough we can expand further on our reasoning in our stand up meeting.

File: hc/front/tests/test_add_channel.py (hunk 2 above)
Comment (georgreen) [reply hc/front/tests/test_add_channel.py]: After looking at it, I think logging in once is enough and we don't need line 51 in the loop, we will move it out of the loop.

File: hc/front/tests/test_add_pushbover.py (hunk 7 above)
Comment (collinmutembei) [reply hc/front/tests/test_add_pushbover.py]: 🆒

**Comment by collinmutembei** [comment]
Looks good. Add the PR description and I'll merge.

**Review by collinmutembei — APPROVED** [review]

**Pull request merged**